app/src/ui/toolbar/branch-dropdown.tsx

```diff
@@ -48,10 +77,99 @@ export class BranchDropdown extends React.Component<IBranchDropdownProps, {}> {
         defaultBranch={branchesState.defaultBranch}
         dispatcher={this.props.dispatcher}
         repository={this.props.repository}
+        account={this.props.account}
+        selectedTab={this.props.selectedTab}
+        pullRequests={this.state.pullRequests}
       />
     )
   }
 
+  public componentDidMount() {
+    if (enablePreviewFeatures()) {
+      this.updatePullRequests(this.props)
+
+      this.refeshPullRequestTimerId = window.setInterval(
+        () => this.updatePullRequests(this.props),
+        RefreshPullRequestInterval
+      )
+    }
+  }
+
+  public componentWillReceiveProps(nextProps: IBranchDropdownProps) {
+    if (
+      enablePreviewFeatures() &&
+      (this.props.account !== nextProps.account ||
+        this.props.repository !== nextProps.repository)
+    ) {
+      this.setState({ pullRequests: null })
+      this.updatePullRequests(nextProps)
+    }
+  }
+
+  public componentWillUnmount() {
+    const timerId = this.refeshPullRequestTimerId
+    if (timerId) {
+      window.clearInterval(timerId)
+      this.refeshPullRequestTimerId = null
+    }
+  }
+
+  private async fetchPullRequests(
+    account: Account,
+    repository: GitHubRepository
+  ): Promise<ReadonlyArray<IPullRequest> | null> {
+    const api = API.fromAccount(account)
+    try {
+      const pullRequests = await api.fetchPullRequests(
+        repository.owner.login,
+        repository.name,
+        'open'
+      )
+
+      const pullRequestsWithStatus: Array<IPullRequest> = []
+      for (const pr of pullRequests) {
+        try {
+          const state = await api.fetchCombinedRefStatus(
+            repository.owner.login,
+            repository.name,
+            pr.head.sha
+          )
+
+          pullRequestsWithStatus.push({
+            ...pr,
+            state,
+            created: new Date(pr.created_at),
+          })
+        } catch (e) {
```

Comment:
Do we care about logging whatever this error is?

Reply:
Nah, this is just filler code until we use the store.